src/Trains.Emoji/Program.cs

```diff
@@ -10,80 +11,141 @@ namespace Trains.Emoji
+                || string.Equals(actual, "/" + expected, StringComparison.OrdinalIgnoreCase)
+                || string.Equals(actual, "-" + expected, StringComparison.OrdinalIgnoreCase);
+        }
+
+        public EmojiDrawer(IRenderer<Tree> tree, TrackRenderer track, BridgeRenderer bridge, IRenderer<Train> train, IPixelMapper pixelMapper)
         {
             _tree = tree;
             _track = track;
+            _bridge = bridge;
             _train = train;
             _pixelMapper = pixelMapper;
         }
 
-        public void Save(int imageSize)
+        public void Save(IEnumerable<int> imageSizes)
         {
-            _pixelMapper.AdjustGameScale(imageSize / 40.0f);
-
-            if (!Directory.Exists(FolderName))
-            {
-                Directory.CreateDirectory(FolderName);
-            }
-            for (int i = 0; i < 3; i++)
-            {
-                Draw("tree" + i, x => _tree.Render(x, new Tree()));
-            }
-            foreach (TrackDirection direction in (TrackDirection[])Enum.GetValues(typeof(TrackDirection)))
+            if (Directory.Exists(BaseFolderName))
             {
-                Draw("track" + direction, x => _track.Render(x, new Track() { Direction = direction }));
+                Directory.Delete(BaseFolderName, true);
             }
-            foreach ((TrackDirection direction, Track track) in ((TrackDirection[])Enum.GetValues(typeof(TrackDirection)))
-                                                    .Select(x => (Direction: x, Track: new Track() { Direction = x, AlternateState = true }))
-                                                    .Where(x => x.Track.HasAlternateState()))
+            foreach (int imageSize in imageSizes)
             {
-                Draw("trackAlt" + direction, x => _track.Render(x, track));
-            }
-            for (int i = 0; i < 6; i++)
-            {
-                DrawTrain($"train{i}Up", TrackDirection.Vertical, 270);
-                DrawTrain($"train{i}Down", TrackDirection.Vertical, 90);
-                DrawTrain($"train{i}Left", TrackDirection.Horizontal, 180);
-                DrawTrain($"train{i}Right", TrackDirection.Horizontal, 0);
+                string folderName = imageSizes.Count() > 1 ? Path.Combine(BaseFolderName, imageSize.ToString()) : BaseFolderName;
+                IPixelMapper pixelMapper = _pixelMapper.Snapshot();
+                pixelMapper.AdjustGameScale(imageSize / 40.0f);
+
+                if (!Directory.Exists(folderName))
+                {
+                    Directory.CreateDirectory(folderName);
+                }
+                for (int i = 0; i < s_numberOfTreesToDraw; i++)
+                {
+                    Draw("tree" + i, folderName, pixelMapper, x => _tree.Render(x, new Tree() { Seed = i }));
+                }
+                foreach (TrackDirection direction in (TrackDirection[])Enum.GetValues(typeof(TrackDirection)))
```

Comment:
Wrap lines 84-95 in a loop and do it with both just the track, and bridge+track?

Reply:
Lots more clean up to do for sure, but it was late 😛